data_steward/cdr_cleaner/cleaning_rules/null_invalid_foreign_keys.py

```diff
@@ -21,84 +29,215 @@
-
-            join_expression = []
-            for key in FOREIGN_KEYS_FIELDS:
-                if key in foreign_keys_flags:
-                    if key == 'person_id':
-                        table_alias = cdr_consts.PERSON_TABLE_NAME
-                    else:
-                        table_alias = _mapping_table_for(
-                            '{x}'.format(x=key)[:-3])
-                    join_expression.append(
-                        LEFT_JOIN.format(dataset_id=dataset_id,
-                                         prefix=key[:3],
-                                         field=key,
-                                         table=table_alias))
-
-            full_join_expression = " ".join(join_expression)
-            query = dict()
-            query[cdr_consts.QUERY] = INVALID_FOREIGN_KEY_QUERY.format(
-                cols=cols,
-                table_name=table,
-                dataset_id=dataset_id,
-                project=project_id,
-                join_expr=full_join_expression)
-            query[cdr_consts.DESTINATION_TABLE] = table
-            query[cdr_consts.DISPOSITION] = bq_consts.WRITE_TRUNCATE
-            query[cdr_consts.DESTINATION_DATASET] = dataset_id
-            queries_list.append(query)
-    return queries_list
+                    table_alias = self.get_mapping_table(
+                        '{x}'.format(x=key)[:-3])
+                join_expression.append(
+                    LEFT_JOIN.render(dataset_id=self.dataset_id,
+                                     prefix=key[:3],
+                                     field=key,
+                                     table=table_alias))
+        return ' '.join(join_expression)
+
+    def get_query_specs(self):
+        """
+        This method gets the queries required to make invalid foreign keys null
+
+        :return: a list of queries
+        """
+        queries_list = []
+        sandbox_queries_list = []
+
+        for table in self.get_affected_tables():
+            if self.has_foreign_key(table):
+                cols = self.get_col_expression(table)
+                join_expression = self.get_join_expression(table)
+
+            invalid_foreign_key_query = {
+                cdr_consts.QUERY:
+                    INVALID_FOREIGN_KEY_QUERY.render(cols=cols,
+                                                     table_name=table,
+                                                     dataset_id=self.dataset_id,
+                                                     project_id=self.project_id,
+                                                     join_expr=join_expression),
+                cdr_consts.DESTINATION_TABLE:
+                    table,
+                cdr_consts.DESTINATION_DATASET:
+                    self.dataset_id,
+                cdr_consts.DISPOSITION:
+                    bq_consts.WRITE_TRUNCATE
+            }
+
+            queries_list.append(invalid_foreign_key_query)
+
+            sandbox_query = {
```

Comment:
`sandbox_query` should be created before `invalid_foreign_key_query` because we want to sandbox records before `invalid_foreign_key_query` updates the content of the table.